bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/TouchWandAlarmSensorHandler.java

```diff
@@ -0,0 +1,123 @@
+            TouchWandUnitDataAlarmSensor sensor = (TouchWandUnitDataAlarmSensor) unitData;
+            updateBatteryLevel(sensor);
+            updateIllumination(sensor);
+            updateChannelLeak(sensor);
+            updateChannelDoorWindow(sensor);
+            updateChannelMotion(sensor);
+            updateChannelTemperature(sensor);
+        } else {
+            logger.warn("updateTouchWandUnitState incompatible TouchWandUnitData instance");
+        }
+    }
+
+    @Override
+    void touchWandUnitHandleCommand(Command command) {
+    }
+
+    void updateBatteryLevel(TouchWandUnitDataAlarmSensor unitData) {
+        Integer battLevel = unitData.getCurrStatus().getBatt();
+        updateState(CHANNEL_BATTERY_LEVEL, new DecimalType(battLevel));
+        int lowThreshold = isBatteryLow ? BATT_LEVEL_LOW + BATT_LEVEL_LOW_HYS : BATT_LEVEL_LOW;
+        boolean lowBattery = (battLevel <= lowThreshold);
+        updateState(CHANNEL_BATTERY_LOW, OnOffType.from(lowBattery));
+        isBatteryLow = lowBattery;
+    }
+
+    void updateIllumination(TouchWandUnitDataAlarmSensor unitData) {
+        for (Sensor sensor : unitData.getCurrStatus().getSensorsStatus()) {
+            if (sensor.type == SENSOR_TYPE_LUMINANCE) {
+                updateState(CHANNEL_ILLUMINATION, new QuantityType<Illuminance>(String.valueOf(sensor.value)));
+            }
+        }
+    }
+
+    void updateChannelLeak(TouchWandUnitDataAlarmSensor unitData) {
+        for (BinarySensorEvent bSensor : unitData.getCurrStatus().getbSensorsStatus()) {
+            if (bSensor.sensorType == SENSOR_TYPE_LEAK) {
+                boolean isLeak = bSensor.sensor.state;
+                updateState(CHANNEL_LEAK, OnOffType.from(isLeak));
+            }
+        }
+    }
+
+    void updateChannelDoorWindow(TouchWandUnitDataAlarmSensor unitData) {
+        for (BinarySensorEvent bSensor : unitData.getCurrStatus().getbSensorsStatus()) {
+            if (bSensor.sensorType == SENSOR_TYPE_DOOR_WINDOW) {
+                boolean isOpen = bSensor.sensor.state;
+                OpenClosedType myOpenClose;
+                myOpenClose = isOpen ? OpenClosedType.OPEN : OpenClosedType.CLOSED;
+                updateState(CHANNEL_DOORWINDOW, myOpenClose);
+            }
+        }
+    }
+
+    void updateChannelMotion(TouchWandUnitDataAlarmSensor unitData) {
+        for (BinarySensorEvent bSensor : unitData.getCurrStatus().getbSensorsStatus()) {
+            if (bSensor.sensorType == SENSOR_TYPE_MOTION) {
+                boolean hasMotion = bSensor.sensor.state;
+                updateState(CHANNEL_MOTION, OnOffType.from(hasMotion));
+            }
+        }
+    }
+
+    void updateChannelTemperature(TouchWandUnitDataAlarmSensor unitData) {
+        for (Sensor sensor : unitData.getCurrStatus().getSensorsStatus()) {
+            if (sensor.type == SENSOR_TYPE_TEMPERATURE) {
+                updateState(CHANNEL_TEMPERATURE, new QuantityType<Temperature>(String.valueOf(sensor.value)));
```

Comment:
Please don't create a QuantityType from a string. If you know the type of unit the device uses you should specify it as the second argument in the QuantityType constructor. For temperature most devices use `SIUnits.CELCIUS`.